aws/signer/v4/v4.go

```diff
@@ -324,7 +325,7 @@ func (v4 Signer) signWithBody(r *http.Request, body io.ReadSeeker, service, regi
 	}
 
 	var err error
-	ctx.credValues, err = v4.Credentials.Retrieve()
+	ctx.credValues, err = v4.Credentials.Retrieve(context.Background())
```

Comment:
The `signWithBody` method, (and its callers) will need to be updated to pass in the context like was done with v2's signer. 

```go
func (v4 Signer) Sign(ctx, context.Context, r *http.Request, body io.ReadSeeker, service, region string, signTime time.Time) (http.Header, error) {
}
```

Reply:
Updated in #461